app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SyncDataRepository.kt

```diff
@@ -0,0 +1,48 @@
+package com.leoart.koreanphrasebook.ui.splash
+
+import android.content.Context
+import com.leoart.koreanphrasebook.data.repository.*
+import com.leoart.koreanphrasebook.ui.chapters.category.CategoriesRepository
+import com.leoart.koreanphrasebook.ui.dialogs.dialog.DiealogRepository
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+
+class SyncDataRepository(context: Context) {
+
+    private val repositories = ArrayList<RefreshableRepository>()
+
+    init {
+        repositories.add(AlphabetRepository(context))
+        repositories.add(ChaptersRepository(context))
+        repositories.add(DialogsRepository(context))
+        repositories.add(DictionaryRepository(context))
+        repositories.add(PhraseRepository(context))
+        repositories.add(CategoriesRepository(context))
+        repositories.add(DiealogRepository(context))
+    }
+
+    fun isSyncNeeded(): Single<Boolean> {
+        val checks = repositories.map { it.isEmpty() }
+        return Single.create<Boolean> { singleEmitter ->
+            Single.merge(checks).toList()
+                    .observeOn(Schedulers.io())
+                    .subscribe({
+                        DataInfoRepository.getInstance().updateData(it)
```

Comment:
@greenhorn31 why do you update data in a function called isSyncNeeded()?
maybe rename it to sync()?

please check what is side effects in programming and how to avoid it.

Reply:
@khrystynapochynok it is not sync, we update data in shared preference to know if sync needed for no network state